File: sgilt-core/src/main/java/net/franzka/sgilt/core/prestataire/dto/PrestataireSearchResponseDto.java
package net.franzka.sgilt.core.prestataire.dto;

import java.util.List;
import java.util.Map;

/**
 * DTO de réponse pour la recherche de prestataires.
 * countsByCategory : compteurs par categoryKey (+ "all")
 * subcatCounts     : compteurs par UUID de sous-catégorie (uniquement pour la catégorie active)
 */
public record PrestataireSearchResponseDto(
        List<PrestataireCardDto> results,
        Map<String, Long> countsByCategory,
        Map<String, Long> subcatCounts
) {}

File: sgilt-front/app/data/prestataire/dto/PrestataireSearchResponseDto.ts
/**
 * DTO — contrat API pour la réponse de recherche de prestataires
 */
import type { PrestataireCardDto } from './PrestataireCardDto'

export interface PrestataireSearchResponseDto {
  results: PrestataireCardDto[]
  countsByCategory: Record<string, number>
  subcatCounts: Record<string, number>
}
